modules/apps/object/object-dynamic-data-mapping/src/main/resources/META-INF/resources/Autocomplete.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {useIsMounted} from '@liferay/frontend-js-react-web';
+import {Autocomplete as BaseAutocomplete} from 'frontend-js-components-web';
+import {debounce} from 'frontend-js-web';
+import React, {useState} from 'react';
+
+function getData(apiUrl, query, page, pageSize) {
+	const url = new URL(apiUrl, Liferay.ThemeDisplay.getPortalURL());
+
+	if (query) {
+		url.searchParams.append('search', query);
+	}
+
+	if (page) {
+		url.searchParams.append('page', page);
+	}
+
+	if (pageSize) {
+		url.searchParams.append('pageSize', pageSize);
+	}
+
+	return Liferay.Util.fetch(url, {
+		headers: new Headers({
+			Accept: 'application/json',
+			'Accept-Language': Liferay.ThemeDisplay.getBCP47LanguageId(),
+			'Content-Type': 'application/json',
+		}),
+	}).then((data) => data.json());
+}
+
+function Autocomplete({
+	apiURL,
+	initialLabel,
+	initialValue,
+	inputName,
+	itemsKey,
+	itemsLabel,
+	value,
+}) {
+	const [query, setQuery] = useState(initialLabel || '');
+
+	const [selectedItem, updateSelectedItem] = useState(initialValue || value);
+	const [items, updateItems] = useState(null);
```

Comment:
Is this where [Require that React.useState returned arguments has consistent naming pattern](https://github.com/liferay/liferay-frontend-projects/issues/655) comes from? 😁 

Are you planning on fixing this later in the process of the PR, or wait for the rule to be enforced?

Reply:
@jbalsas I was waiting to enforce is until the rule is in, and thats sort of the pattern I am following for things right now. If I can create an eslint rule for it, I don't bother leaving that in my review yet because I that way we can just make that change all at once.

For this code specifically, I just copied it from the old implementation and didn't get the standards too much thought at the time, but as I go through it now I am updating it and creating any issues for rules we need for eslint.